### DISCUSSION TITLE
CustomTkinter

### DISCUSSION THREAD

**Discussion by PyCryptic** [TomSchimansky/CustomTkinter#2544] [Q&A]
have the following problem:
I want to access a button inside a grid layout, to change its colour later in the code.
What is the best way to implement accessing the buttons from the grid?

**Comment by dipeshSam** [comment]
Use `.grid_slaves()` method of any widget. This method returns a list of children widgets located at the specified grid location using row and column values.

Example code:
``` Python
from customtkinter import CTk, CTkFrame, CTkButton

global frame


def on_click():
    button = frame.grid_slaves(row=0, column=0)[0]
    
    active = True if button.cget("text") != "Stop" else False      
    
    button.configure(text="Stop" if active else "Start",
                     fg_color="brown" if active else "green",
                     hover_color="darkred" if active else "darkgreen")
    
       

if __name__ == "__main__":
    app = CTk()    
    app.geometry("900x600")

    frame = CTkFrame(app, fg_color="gray25")
    frame.place(relx=0.5, anchor="center", rely=0.5, relheight=0.5, relwidth=0.5)
    
    CTkButton(frame,
              text="Start",
              command=on_click,
              fg_color="green",
              hover_color="darkgreen").grid(row=0, column=0, padx=20, pady=20)
    


    app.mainloop()
    
    
```

**Reply by PyCryptic** [reply]
Thank you for your help! The explanation and example you gave made everything much clearer and I was able to understand the usage of the .grid_slaves() method perfectly. Your help was invaluable and I really appreciate the time you took to explain it. This will definitely make my work with CustomTkinter more efficient. Thank you again, it really helped me a lot!
